packages/flutter/lib/src/rendering/error.dart

```diff
@@ -50,6 +51,9 @@ class RenderErrorBox extends RenderBox {
 
   /// The message to attempt to display at paint time.
   final String message;
+  final double systemTopPadding = Zone.current['systemTopPadding'] == null
```

Comment:
> Would you move all of dart:ui into a Zone?

We only have one static, so there isn't much else to move.  :)

> Also, in addition to the loss of strong typing (by which I specifically mean that the analyzer can't verify that code reading out of zones in this way is correct), you lose dartdocs and autocompletion.

I don't quite understand this part.  I might need to ask you to explain in person.

> unless you mean specifically for things we are forcing them to use zones

Yes, I meant specifically for the `window` static.

Ok, I'm happy with the `@protected` member function approach you suggest.  If we find ourselves doing that in many places, we can re-evaluate whether we want a more centralized solution.